packages/legacy/borsh-schema/package.json

```diff
@@ -0,0 +1,90 @@
+{
+    "name": "@solana/borsh-schema",
```

Comment:
The way I think about this is that it _is_ and always will be the Borsh schema, it's just that it will become irrelevant at some point, and depended on by nothing much.

Reply:
I think standalone packages makes sense for `ed25519` and `borsh`, but anything Solana should be legacy-prefixed.

I also think we should keep in mind we have to maintain these if they're not marked legacy. Until deprecated.

Reply:
I don't particularly agree that `borsh-schema` is and always will be the Borsh schema. It is the way we _used_ to encode and decode data before we added `codecs`. My worry is without the `legacy-` prefix, users will be confused as they'll see this new `borsh-schema` package and think this is the new recommended way of encoding things.

I don't know enough about the `ed25519` package but my heuristic here would be: could we, in theory, use `@solana/ed25519` in conjunction with the other packages of the new web3.js. If it conflicts with some of the current packages whatsoever, I think it'd benefit from being prefixed with `legacy-`.

Reply:
I think even with `ed25519` I'm concerned about people thinking this is the recommended way to sign, verify and compute program derived addresses. Ideally we should recommend people to use the async method provided by the CryptoKeyPair API instead.

If we're saying `ed25519` is perfectly compatible with the new web3.js then I might start using this instead so I can generate my PDA synchronously and avoid having to generate two functions haha.

Reply:
> If we're saying `ed25519` is perfectly compatible with the new web3.js then I might start using this instead so I can generate my PDA synchronously and avoid having to generate two functions haha.

Shhhh!

Ok, I have to admit I believe people will 100% do exactly that. Even just calling it `ed25519-userspace` wouldn't help. Only `-legacy` would deter.

Do we roll this into `@solana/keys@1.x`?

Reply:
> Do we roll this into @solana/keys@1.x?

Mmmm… I thought there was a reason I didn't do this, but it looks like literally everything that pulls in `@solana/ed255519` _also_ pulls in `@solana/keys@1` so… I guess?

Reply:
Ugh. This means that all the `generatePrivateKey` kind of methods _have_ to be exported from `@solana/keys` which I wanted to avoid. It also makes it more annoying to _prevent_ those from being exported from `@solana/web3.js` (I have to enumerate all of the exports of `@solana/keys` rather than being able to `export * from '@solana/keys'`.

I'll sleep on it.

Reply:
I wonder if it's enough to just straight up add an `npm deprecate @solana/ed25519@x.y.z` command to the build script. Would make it literally not show up in npm (I think) but wouldn't prevent `@solana/web3.js@1` from depending on it. Thoughts?

Reply:
I personally wouldn't use such a nice npm package name on the precious `@solana` namespace only to deprecate it immediately. What if we really need this in the future? Also, what's wrong with prefixing all these libraries with `legacy-`? I'm struggling to understand why we go to such lengths (like keys@v1) to avoid doing that.